### PR TITLE
[Core] Add worker resource info to runtime env

### LINKED ISSUE #18617: Add worker resource info into CreateRuntimeEnvRequest
### Describe your feature request

Address https://github.com/ray-project/ray/pull/18185 . I am trying to put cgroup management into runtime_env agent. 

Worker resource info is required to setup cgroup. The worker resource info has been serialized to `allocated_instances_serialized_json`.  How about  adding  worker resource info into CreateRuntimeEnvRequest?  

@edoakes

### PR DESCRIPTION
**Pull request by chenk008**
## Why are these changes needed?

We want to add resource limit for worker https://github.com/ray-project/ray/issues/17596.  

This is the first pull request. This change add worker allocated resource instance into `CreateRuntimeEnvRequest`.

## Related issue number

Close https://github.com/ray-project/ray/issues/18617

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/runtime_env/cgroup.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+default_logger = logging.getLogger(__name__)
+
+
+class CgroupManager:
+    def __init__(self):
+        pass
+
+    def setup(self,
+              allocated_resource: dict,
```
Comment (edoakes) [inline python/ray/_private/runtime_env/cgroup.py:14]: Can you make this a kwarg so it's optional given that it won't be used by most plugins? Also, please add type annotattions for the dictionary (keys and values)

File: src/ray/protobuf/runtime_env_agent.proto [hunk 2]
```diff
@@ -21,6 +21,8 @@ import "src/ray/protobuf/agent_manager.proto";
 message CreateRuntimeEnvRequest {
   string serialized_runtime_env = 1;
   bytes job_id = 2;
+  // Allocated resource instances.
```
Comment (edoakes) [inline src/ray/protobuf/runtime_env_agent.proto:24]: more descriptive comment about what this dictionary contains please

File: python/ray/_private/runtime_env/cgroup.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import logging
```
Comment (simon-mo) [inline python/ray/_private/runtime_env/cgroup.py:1]: remove this file and submit it as part of the PR for plugin?

**Review by simon-mo — APPROVED** [review]
LGTM. Please address @edoakes's comment. Adding more metadata to help runtime env plugin is great!

cc @architkulkarni when you are making the plugin structure, make sure to put allocated_resource_instances in the context dictionary for plugin so @chenk008 can use that information for cgroup plugin.

File: python/ray/_private/runtime_env/cgroup.py (hunk 3 above)
Comment (chenk008) [reply python/ray/_private/runtime_env/cgroup.py:1]: Thanks for your suggestions, I have deleted it.

File: src/ray/protobuf/runtime_env_agent.proto (hunk 2 above)
Comment (chenk008) [reply src/ray/protobuf/runtime_env_agent.proto]: Thanks for your suggestion. I have add some comment for it , PTAL.

**Pull request merged**